### DISCUSSION TITLE
UseFieldArray, shouldUnregister:false and reset

### DISCUSSION THREAD

**Discussion by spirit-team** [react-hook-form/react-hook-form#3034] [Q&A]
Hi,

We have a form in which we can create several request (same logic, each in a tab). It's fine to load defaultValues and having the right values in each tab. And when we submit the form, we have all tabs values.

But when we try to reset the form(simulating getting data from API), only current tab data is in the values. if we navigate in unselected tabs, right data appears in the values.  (try to comment line 66 - reset(props.initialValues);)

Is there a way to reset useFieldArray like in the initial defaultValues?

I hope the CSB is good enough:
https://codesandbox.io/s/react-hook-from-async-set-form-values-forked-pso6l?file=/src/form.js

p.s. I've tried css hidden tabs, but with severals tabs (50+), rendering is very slow..

**Comment by bluebill1049** [comment]
your example is confusing to me, so i did a quick one myself: https://codesandbox.io/s/blue-browser-5ld70?file=/src/App.js

let me know if i am missing something.

**Reply by spirit-team** [reply]
Your example is way simpler than mine ;)

I've adapted it to show my issue.

https://codesandbox.io/s/gracious-minsky-i3t4u?file=/src/App.js

when it's load the defaultValues, every fields of the field array appears in the submit. but when I reset the data with other value, it shows only the selecteds steps (not all data in the reset values).

I was thinking using reset to update form data with data from remote API.

**Reply by n-shkutov-awtg** [reply]
@bluebill1049 if you change version of react-hook-form from `6.8.6` to `6.11.5` and push reset you will see the issue.

**Reply by spirit-team** [reply]
@n-shkutov-awtg Thx for the heads up.
It does indeed seem to break with version 6.11.0

**Reply by bluebill1049** [reply]
@n-shkutov-awtg you refer to my example or @spirit-team's. 

@spirit-team it's not about how complex the example is, if you can reproduce the issue a simpler (less side effect) one will help more.

**Reply by spirit-team** [reply]
Nevermind, it seems to work. Not quite sure what was wrong with the original sandbox, but it works in this one.

https://codesandbox.io/s/happy-chaplygin-vwtyv?file=/src/App.js
